### ISSUE TITLE
Add block macro extension example to user manual

### ISSUE THREAD

**Issue by graphitefriction** [asciidoctor/asciidoctor.org#167]
Add the example and explanations provided in Issue [#734](https://github.com/asciidoctor/asciidoctor/issues/734) to the extensions part of the user manual.

Also include path resolution utilities per @mojavelinux.

**Comment by jaredmorgs** [comment]
This goes into the http://asciidoctor.org/docs/user-manual/#extensions section as a new sub-section

At the end of this section, there is reference to https://github.com/asciidoctor/asciidoctor-extensions-lab

I wonder if the example extension should be added there instead of the guide? Or should we really be documenting the "best" example out of each extension lab group and then directing users to other examples in the extension lab?

**Comment by mojavelinux** [comment]
>  should we really be documenting the "best" example out of each extension lab group and then directing users to other examples in the extension lab?

Exactly. I like that strategy. And we can even use a remote include to pull in the source code from the extension lab so that we don't duplicate it. However, we will want to document that in the source file so that someone editing it knows it is being used. We could also link to a specific version of the file.

**Comment by mojavelinux** [comment]
The user manual does have a block processor example. See http://asciidoctor.org/docs/user-manual/#block-macro-processor-example. However, it doesn't explain how to insert a new block and specify which substitutions should be applied. So the focus should be on creating new AST nodes in an extension and inserting them into the tree. We should reference the API http://www.rubydoc.info/gems/asciidoctor/Asciidoctor/Block#initialize-instance_method
